File: backend/app/com/skewwhiffy/auraltester/services/IntervalService.scala
package com.skewwhiffy.auraltester.services

import com.skewwhiffy.auraltester.clefs.Clef
import com.skewwhiffy.auraltester.notes.{Interval, Note}
import com.skewwhiffy.auraltester.scales.IntervalNotes

import scala.util.chaining.scalaUtilChainingOps

class IntervalService {
  def getInterval(clef: Clef, note: Note, interval: Interval): IntervalNotes = {
    clef.getNoteNearBottom(note).pipe(it => new IntervalNotes(it, interval))
  }

}

File: backend/src/main/kotlin/com/skewwhiffy/service/IntervalService.kt
package com.skewwhiffy.service

import com.skewwhiffy.notation.model.clef.Clef
import com.skewwhiffy.notation.model.interval.Interval
import com.skewwhiffy.notation.model.note.IntervalNotes
import com.skewwhiffy.notation.model.note.Note
import org.springframework.stereotype.Service

@Service
class IntervalService {
  fun getInterval(clef: Clef, note: Note, interval: Interval): IntervalNotes =
    IntervalNotes(clef.getNoteNearBottom(note), interval)
}
